File: src/main/java/com/github/zametki/model/Identifiable.java
package com.github.zametki.model;

/**
 * Database entity with ID. Not serializable and can't not be stored in Wicket sessions.
 */
public class Identifiable<T extends AbstractId> {
    public T id;

    public Identifiable() {
    }

    public String toString() {
        return getClass().getSimpleName() + "[id:" + id + "]";
    }
}

File: src/main/java/com/github/zametki/model/Identifiable.kt
package com.github.zametki.model

/**
 * Database entity with ID. Not serializable and can't not be stored in Wicket sessions.
 */
open class Identifiable<T : AbstractId> {
    //TODO: make not nullable
    var id: T? = null

    override fun toString() = javaClass.simpleName + "[id:" + id + "]"
}
